lib/src/documents.dart

```diff
@@ -0,0 +1,63 @@
+import 'package:typesense/src/api_call.dart';
+import 'package:typesense/src/collections.dart';
+import 'dart:convert';
+
+class Documents {
+  final ApiCall _apicall;
+  final String _collectionName;
+  static const String RESOURCEPATH = '/documents';
+
+  Documents(String collectionName, ApiCall apicall)
+      : _apicall = apicall,
+        _collectionName = collectionName;
+
+  Future<Map<String, dynamic>> create(Map<String, dynamic> document,
+      {Map<String, dynamic> options = const {}}) async {
+    Map<String, dynamic> ops = new Map<String, dynamic>.from(options)
+      ..addAll({"action": "create"});
+    return await _apicall.post(_endPoint(),
+        bodyParameters: document, queryParams: ops);
+  }
+
+  Future<Map<String, dynamic>> upsert(Map<String, dynamic> document,
+      {Map<String, dynamic> options = const {}}) async {
+    Map<String, dynamic> ops = new Map<String, dynamic>.from(options)
+      ..addAll({"action": "upsert"});
+    return await _apicall.post(_endPoint(),
+        bodyParameters: document, queryParams: ops);
+  }
+
+  Future<Map<String, dynamic>> update(Map<String, dynamic> document,
+      {Map<String, dynamic> options = const {}}) async {
+    Map<String, dynamic> ops = new Map<String, dynamic>.from(options)
+      ..addAll({"action": "update"});
+    return await _apicall.post(_endPoint(),
+        bodyParameters: document, queryParams: ops);
+  }
```

Comment:
`update` function should call `ApiCall.patch` function instead according to the [docs](https://typesense.org/docs/0.20.0/api/documents.html#update-a-document)

Reply:
Just checked the docs again. I got `Documents` confused with `Document`. All good!